### PR TITLE
Don't fetch nOpt another time

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#1981 

# Overview

- [x] Re-use the existing `nOpt` instead of fetching it again

# Comments

### CODE REVIEW

File: lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs [hunk 1]
```diff
@@ -272,11 +276,11 @@ data PoolDbData = PoolDbData
 combineDbAndLsqData
     :: forall m. (Monad m)
     => TimeInterpreter m
-    -> ProtocolParameters
+    -> Int -- ^ nOpt; desired number of pools
```
Comment (Anviking) [inline lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs:279]: Do we need a `newtype DesiredNumberOfPools`? 

I'm not sure the need is really there yet, and making all the other packages compile again can be slightly annoying, so I didn't do it.

Comment (paweljakubas) [reply lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs:279]: hmm, I do not see this newtype: 
```
pawel@pure:~/Work/IOHK/cardano-wallet (master)$ ag DesiredNumberOfPools
lib/core/src/Cardano/Wallet/DB/Sqlite/TH.hs
168:    protocolParametersDesiredNumberOfPools  Word16          sql=desired_pool_number

lib/core/src/Cardano/Wallet/DB/Sqlite.hs
450:        addColumn conn True (DBField ProtocolParametersDesiredNumberOfPools) value
```

**Review by paweljakubas — APPROVED** [review]
one less rountrip. lgtm !

**Pull request merged**